### PR TITLE
Cross-platform build

### PR DESCRIPTION
**Pull request by falkTX**
Now building and running on Linux :tada: 

image

Sadly it crashes anytime a JIT operation is triggered, so it is not as useful.
The `memofs` for any call is just very suspiciously wrong.
I couldn't find why, sorry.

Allowing code to build is a good first step though, so here you go :)
In theory this should allow to build on macOS too, but I did not test this.

### CODE REVIEW

**Comment by jarikomppa** [comment]
Heh, this is pretty much exactly as far as I got when I started porting to linux.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/make -f
+# Makefile for FluidPlug #
```
Comment (jarikomppa) [inline Makefile:2]: The what? =)

File: Makefile.mk [hunk 2]
```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/make -f
+# Makefile for DPF #
```
Comment (jarikomppa) [inline Makefile.mk:2]: ..

File: src/eval.cpp [hunk 3]
```diff
@@ -727,14 +727,14 @@ std::string func2call(int func)
 	return s;
 }
 
-WELL512 random;
+WELL512 wrandom;
```
Comment (jarikomppa) [inline src/eval.cpp:730]: this probably should be gRandom, my bad

**Review by jarikomppa — COMMENTED** [review]
some small notes

File: Makefile (hunk 1 above)
Comment (falkTX) [reply Makefile:2]: lol copy&paste silliness, you didn't see anything :rainbow:

File: src/eval.cpp (hunk 3 above)
Comment (falkTX) [reply src/eval.cpp:730]: I can change that, no worries.
It is just that we get a conflict with `random` function, leading to a build failure

**Comment by falkTX** [comment]
just pushed fixes for the little issues you commented on.

**Comment by jarikomppa** [comment]
Thanks!

**Comment by jminor** [comment]
@falkTX I see a very similar crash on macOS. See #3 for details.

**Pull request merged**